### PR TITLE
chore: Small fixes to unit tests and fixed many warnings

### PR DESCRIPTION
**Pull request by mohammedDehairy**
chore: Small fixes to unit tests and fixed many warnings

### CODE REVIEW

File: AdyenTests/Card Tests/CardComponentTests.swift [hunk 1]
```diff
@@ -247,7 +247,7 @@ class CardComponentTests: XCTestCase {
         let method = CardPaymentMethod(type: "bcmc", name: "Test name", fundingSource: .credit, brands: ["any_test_brand_name"])
         let sut = CardComponent(paymentMethod: method,
                                 clientKey: "test_client_key")
-        sut.showsLargeTitle = false
+        sut._showsLargeTitle = false
```
Comment (descorp) [inline AdyenTests/Card Tests/CardComponentTests.swift:250]: I think we can delete those :)

File: AdyenTests/Adyen Tests/Components/Apple Pay/ApplePayComponentTest.swift [hunk 2]
```diff
@@ -69,6 +69,23 @@ class ApplePayComponentTest: XCTestCase {
         waitForExpectations(timeout: 2)
     }
     
+    func testApplePayViewControllerIsResetAfterAuthorization() {
```
Comment (descorp) [inline AdyenTests/Adyen Tests/Components/Apple Pay/ApplePayComponentTest.swift:72]: As I mentioned yesterday - this test case is excessive:

we already have tests for `paymentAuthorizationViewController(didAuthorizePayment:)` and `stopLoading`.

File: AdyenDropIn/Utilities/ComponentManager.swift [hunk 3]
```diff
@@ -80,7 +80,6 @@ internal final class ComponentManager {
             return nil
         }
         
-        cardComponent?.showsLargeTitle = false
```
Comment (descorp) [inline AdyenDropIn/Utilities/ComponentManager.swift:83]: default value is `true`
I believe you have fixed and tested it everywhere?

File: AdyenCard/Utilities/Encryption/Cryptor/AES.swift [hunk 4]
```diff
@@ -232,16 +233,20 @@ extension Cryptor.AES {
...
-                                            encodeBytes: Int,
-                                            counter: UInt,
-                                            pointer: Pointer,
-                                            length: Int,
-                                            blockA: MutableBytes,
-                                            blockS: MutableBytes) {
-                setCounter(blockA, encodeBytes, counter)
-                aesEncrypt(block: blockA, with: key, saveTo: blockS)
-                memxor(pointer, blockS, length)
+            internal struct EncryptXorContext {
```
Comment (descorp) [inline AdyenCard/Utilities/Encryption/Cryptor/AES.swift:236]: I wonder if there is a better way to do this 🤔
It is better than 7 params, tho

File: AdyenDropIn/Utilities/ComponentManager.swift (hunk 3 above)
Comment (mohammedDehairy) [reply AdyenDropIn/Utilities/ComponentManager.swift:83]: in those components, they check if `showsLargeTitle` is true AND `_isDropIn` is false, so this should cover this scenario.

File: AdyenTests/Adyen Tests/Components/Apple Pay/ApplePayComponentTest.swift (hunk 2 above)
Comment (mohammedDehairy) [reply AdyenTests/Adyen Tests/Components/Apple Pay/ApplePayComponentTest.swift:72]: its necessary to test that the `viewController` is nilled after calling stopLoading, right?

File: AdyenTests/Card Tests/CardComponentTests.swift (hunk 1 above)
Comment (mohammedDehairy) [reply AdyenTests/Card Tests/CardComponentTests.swift:250]: don't think so, they are still needed, to setup the test preconditions.

File: AdyenTests/Adyen Tests/Components/Apple Pay/ApplePayComponentTest.swift (hunk 2 above)
Comment (descorp) [reply AdyenTests/Adyen Tests/Components/Apple Pay/ApplePayComponentTest.swift:72]: it is already [tested here](https://github.com/Adyen/adyen-ios/blob/d478372a95e4d798120e42e3385f3e20350f68cc/AdyenTests/Adyen%20Tests/Components/Apple%20Pay/ApplePayComponentTest.swift#L37-L43)

File: AdyenDropIn/Utilities/ComponentManager.swift (hunk 3 above)
Comment (descorp) [reply AdyenDropIn/Utilities/ComponentManager.swift:83]: 👍

**Review by descorp — APPROVED** [review]

File: AdyenTests/Adyen Tests/Components/Apple Pay/ApplePayComponentTest.swift (hunk 2 above)
Comment (mohammedDehairy) [reply AdyenTests/Adyen Tests/Components/Apple Pay/ApplePayComponentTest.swift:72]: ok, I will remove it, but in another PR, is that okay?

**Pull request merged**